src/proto/uprotocol/upayload.rs

```diff
@@ -0,0 +1,68 @@
+/********************************************************************************
+ * Copyright (c) 2023 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ********************************************************************************/
+
+use prost::Message;
+use prost_types::Any;
+use std::slice;
+
+use crate::uprotocol::{Data, UPayload};
+
+impl From<Any> for UPayload {
+    fn from(value: Any) -> Self {
+        let vec = value.encode_to_vec();
+        let len = vec.len() as i32;
```

Comment:
In Rust length is usually represented with an `usize`. Any reason on using an `i32`?

Reply:
The protobuf-generated UPayload type takes i32 for it's length field, that is the reason for the cast.

Reply:
Ok, then to be safe in terms of hidden bugs cause by casting, I would explicitly perform a conversion between `usize` and `i32` with a [TryFrom](https://doc.rust-lang.org/std/primitive.i32.html#impl-TryFrom%3Cusize%3E-for-i32). That would be able to catch the case of the value length to be greater of `i32::MAX`.

This leads also to the fact that the `From<Any>` trait implentation should be a `TryFrom<Any>` since it could fail?

Reply:
TryFrom is not possible unfortunately, as that is already implemented as part of the prost-generated code. But I will adopt your point about doing an error-aware cast.